### PR TITLE
Add EIP-7549 to Electra

### PR DESCRIPTION
**Pull request by ralexstokes**
~builds on top of #3615~

~rebased #3615 against dev @ f8cbd8d2e3c7d046436b290a9cc5e29e3ec78989.~

merges EIP-7549 into `electra`

## TODO

- [ ] add warning/note about assembling `AttesterSlashings` with pre-Electra attestations?
- [ ] how to handle Deneb `Attestation`s that can be included in Electra but won't be compatible with the new format?
  * this commit has one solution but enforces about 1 epoch of inclusion delay: d35b4091fb34339517ae5c7bce9a786298954b79
  * one suggestion: clients broadcast both formats in the last deneb epoch, with care taken to not produce slashable messages
  * another suggestion: extend block with extra field for the pre-electra attestations and just allow the deneb style during the first epoch of electra (otherwise validate this new list is empty in electra)
- [x] merge this in: https://github.com/ethereum/consensus-specs/pull/3660

### CODE REVIEW

File: specs/electra/beacon-chain.md [hunk 1]
```diff
@@ -232,6 +386,35 @@ def process_deposit_receipt(state: BeaconState, deposit_receipt: DepositReceipt)
...
 ```
 
+#### New `process_execution_layer_exit`
+
+*Note*: This function is new in Electra:EIP7002.
+
+```python
+def process_execution_layer_exit(state: BeaconState, execution_layer_exit: ExecutionLayerExit) -> None:
+    validator_pubkeys = [v.pubkey for v in state.validators]
+    validator_index = ValidatorIndex(validator_pubkeys.index(execution_layer_exit.validator_pubkey))
```
Comment (g11tech) [inline specs/electra/beacon-chain.md:396]: should there be a check to see if pubkey is in pubkeys and silent return if not?

Comment (ralexstokes) [reply specs/electra/beacon-chain.md:396]: yep! everything coming from the EL should assume to be untrusted and so we should validate it, and if something doesn't line up should just skip over processing this particular `execution_layer_exit`

python will error here, which following the spec semantics would imply the block is invalid, giving anyone a cheap way to dos the chain -- not what we want!

Comment (ralexstokes) [reply specs/electra/beacon-chain.md:396]: https://github.com/ethereum/consensus-specs/pull/3660

File: .gitignore [hunk 2]
```diff
@@ -22,6 +22,10 @@ tests/core/pyspec/eth2spec/bellatrix/
 tests/core/pyspec/eth2spec/capella/
 tests/core/pyspec/eth2spec/deneb/
 tests/core/pyspec/eth2spec/electra/
+<<<<<<< HEAD
+=======
+tests/core/pyspec/eth2spec/eip7549/
+>>>>>>> dev
```
Comment (dapplion) [inline .gitignore:28]: warning, non-resolved conflict

File: .gitignore [hunk 3]
```diff
@@ -22,7 +22,10 @@ tests/core/pyspec/eth2spec/bellatrix/
 tests/core/pyspec/eth2spec/capella/
 tests/core/pyspec/eth2spec/deneb/
 tests/core/pyspec/eth2spec/electra/
+<<<<<<< HEAD
+=======
 tests/core/pyspec/eth2spec/eip7549/
+>>>>>>> dev
```
Comment (hwwhww) [inline .gitignore:28]:
```suggestion
```

**Pull request merged**